src/main/kotlin/lotto/model/LottoMachine.kt

```
@@ -0,0 +1,122 @@
+                return manualTicketCount
+            } catch (_: IllegalArgumentException) {
+                println("Enter a valid manual ticket count")
+            }
+        }
+    }
+
+    private fun getManualTickets(manualTicketCount: Int): List<Ticket> {
+        while (true) {
+            try {
+                val manualTicketsString = InputView.inputManualTicketNumbers(manualTicketCount)
+                val manualTickets = ManualTicket.generateManualTickets(manualTicketsString)
+                return manualTickets
+            } catch (_: IllegalArgumentException) {
+                println("Enter valid manual tickets")
+            }
+        }
+    }
+
+    private fun generateAutomaticTickets(
+        amount: Int,
+        manualTicketCount: Int,
+    ): List<Ticket> {
+        val ticketCount = amount / Constants.TICKET_PRICE - manualTicketCount
+        val ticketsList = mutableListOf<Ticket>()
+        repeat(ticketCount) {
+            val ticket = Ticket(generateTicketNumbers())
+            ticketsList.add(ticket)
+        }
+        OutputView.displayTickets(ticketCount, ticketsList, manualTicketCount)
+        return ticketsList
+    }
+
+    private fun generateTicketNumbers(): List<LottoNumber> {
+        return LottoNumber.Companion.takeRandom(Constants.NUMBER_COUNT)
+    }
+
+    private val winStats =
+        mutableMapOf(
+            Rank.FIRST to 0,
+            Rank.SECOND to 0,
+            Rank.THIRD to 0,
+            Rank.FOURTH to 0,
+            Rank.FIFTH to 0,
+            Rank.MISS to 0,
+        )
+
+    private fun calculateStats(
+        tickets: List<Ticket>,
+        winningNumbers: List<LottoNumber>,
+        bonusNumber: LottoNumber,
+    ) {
+        for (ticket in tickets) {
+            val match = checkMatch(ticket, winningNumbers)
+            if (match == 5 && ticket.numbers.contains(bonusNumber)) {
+                winStats[Rank.SECOND] = winStats[Rank.SECOND]!! + 1
```

I saw this `!!` operator in the previous step, but I decided not to comment on it yet.
Now, I think it's a good time for us to think about it :)

Do you know what this operator does?
I wouldn't say it's wrong, but I wanted to know if it were your intention.
Also, what are the alternatives to this approach?

---

There was a complitaion error "Operator call corresponds to a dot-qualified call 'winStats[rank].plus(1)' which is not allowed on a nullable receiver 'winStats[rank]' " So we deiced to use a Not Null Assertion Operator (!!) we used it as we were certain that the value will not be nul, it converts any nullable type to a non-nullable type and throws a Null Pointer Exception if the value is null.
on byte code it looks like 
`winStats.put(rank, ((Number)winStats.get(rank)).intValue() + 1);` and its using .intValue() which cannot receive null value before trying to increment it.

another safer option would be to use Elvis operator (?:) or getOrDefault 
`winStats[rank] = winStats[rank] ?.plus(1) ?: 1`
or
`winStats.put(rank, winStats.getOrDefault(rank, 0) + 1)`

---

Nice!
To simplify, we get to choose if we want to throw the exception or to go on with our logic,
even though, like you said, we're pretty certain that it won't be null. 🤔 
It's just one of those 'machines don't understand us' moments, and I thought of another option :  
```Kotlin
winStats[Rank.SECOND] = winStats.getOrDefault(Rank.SECOND, 0) + 1
```

Your alternatives also work well, and even keeping `!!` isn't a wrong choice.
Nice explanation 👍